### PR TITLE
bonuses

### PR DESCRIPTION
**Pull request by ice-orion**

### CODE REVIEW

File: ios/ice.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>PreviewsEnabled</key>
```
Comment (ice-ares) [inline ios/ice.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings:5]: whats this?

File: src/store/modules/Tokenomics/sagas/handleExtraBonus.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      while (!(yield* isRegistrationComplete())) {
+        yield take('*');
+      }
+
+      yield call(openClaimBonus);
+
+      const extraBonus: SagaReturnType<typeof Api.tokenomics.claimExtraBonus> =
+        yield call(Api.tokenomics.claimExtraBonus, {userId});
+
+      if (!extraBonus.availableExtraBonus) {
```
Comment (ice-ares) [inline src/store/modules/Tokenomics/sagas/handleExtraBonus.ts:37]: this never happens. u get an error, not an empty response

File: src/store/modules/Tokenomics/sagas/handleExtraBonus.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    userIdSelector,
+  );
+
+  try {
+    if (miningSummary?.availableExtraBonus) {
+      while (!(yield* isRegistrationComplete())) {
+        yield take('*');
+      }
+
+      yield call(openClaimBonus);
```
Comment (ice-ares) [inline src/store/modules/Tokenomics/sagas/handleExtraBonus.ts:32]: does this block until I press "Claim now" ?

File: src/store/modules/Tokenomics/utils/openBonusClaimed.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import {openLinkWithInAppBrowser} from '@utils/device';
+import React from 'react';
+import {ImageSourcePropType} from 'react-native';
+
+export const openBonusClaimed = ({claimedBonus}: {claimedBonus: number}) => {
+  let resultResolve: () => void;
+  const resultPromise = new Promise<void>(r => (resultResolve = r));
+
+  let image: ImageSourcePropType;
+  if (claimedBonus <= 130) {
```
Comment (ice-ares) [inline src/store/modules/Tokenomics/utils/openBonusClaimed.tsx:18]: const the values

File: src/store/modules/Tokenomics/utils/openClaimBonus.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      message: <Message text={message} />,
+      warning: (
+        <Touchable
+          onPress={() => openLinkWithInAppBrowser({url: LINKS.BONUSES})}>
+          <Warning text={t('extra_bonus.link')} />
+        </Touchable>
+      ),
+      buttons: [
+        {
+          Icon: <StakeIcon />,
```
Comment (ice-ares) [inline src/store/modules/Tokenomics/utils/openClaimBonus.tsx:45]: can you wrap that stakeIcon and name it differently, I.E> ClaimBonusIcon.

Cuz its confusing to see this here

**Review by ice-atlas — CHANGES_REQUESTED** [review]

File: ios/ice.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings (hunk 1 above)
Comment (ice-orion) [reply ios/ice.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings:5]: removed, added to gitignore

File: src/navigation/Router.tsx [hunk 6]
```diff
@@ -9,46 +8,37 @@ import {WelcomeNavigator} from '@navigation/Welcome';
...
   if (!isAppInitialized) {
     return null; // previously we returned Initialization component but now null is ok since we have animated splash over the navigator
+  } else {
+    navigationReadyResolver();
   }
 
   if (!user) {
     return <AuthNavigator />;
   }
+  console.log('isRegistrationComplete', isRegistrationComplete);
```
[the remark this answers is not in the document]
Comment (ice-orion) [reply src/navigation/Router.tsx:39]: done

File: src/store/modules/Tokenomics/sagas/handleExtraBonus.ts (hunk 2 above)
Comment (ice-orion) [reply src/store/modules/Tokenomics/sagas/handleExtraBonus.ts:37]: removed this block

File: src/store/modules/Tokenomics/sagas/handleExtraBonus.ts (hunk 3 above)
Comment (ice-orion) [reply src/store/modules/Tokenomics/sagas/handleExtraBonus.ts:32]: yes

File: src/store/modules/Tokenomics/utils/openBonusClaimed.tsx (hunk 4 above)
Comment (ice-orion) [reply src/store/modules/Tokenomics/utils/openBonusClaimed.tsx:18]: done

File: src/store/modules/Tokenomics/utils/openClaimBonus.tsx (hunk 5 above)
Comment (ice-orion) [reply src/store/modules/Tokenomics/utils/openClaimBonus.tsx:45]: done, renamed to CoinsStakeIcon

File: src/store/modules/Tokenomics/sagas/handleExtraBonus.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    miningSummarySelector,
+  );
+
+  const userId: ReturnType<typeof userIdSelector> = yield select(
+    userIdSelector,
+  );
+
+  try {
+    if (miningSummary?.availableExtraBonus) {
+      while (!(yield* isRegistrationComplete())) {
```
[the remark this answers is not in the document]
Comment (ice-orion) [reply src/store/modules/Tokenomics/sagas/handleExtraBonus.ts:28]: done

File: src/store/modules/Tokenomics/utils/openClaimBonus.tsx (hunk 5 above)
Comment (ice-ares) [reply src/store/modules/Tokenomics/utils/openClaimBonus.tsx]: 😆 thats even worse. There's nothing related to staking there, thats my point.

Comment (ice-orion) [reply src/store/modules/Tokenomics/utils/openClaimBonus.tsx]: Right, it supposed to be CoinsStack :)
We're dreaming icons by what it is depicted there, so this way you're not restricted by the usage case.

Comment (ice-ares) [reply src/store/modules/Tokenomics/utils/openClaimBonus.tsx]: yea

Comment (ice-orion) [reply src/store/modules/Tokenomics/utils/openClaimBonus.tsx]: renamed

File: fastlane/Matchfile [hunk 8]
```diff
@@ -1,3 +1,5 @@
 # SPDX-License-Identifier: BUSL-1.1
 
 type("appstore")
+
+app_identifier(["io.ice.app.staging", "io.ice.app", "io.ice.app.staging.NotificationService", "io.ice.app.NotificationService"])
```
Comment (ice-ares) [inline fastlane/Matchfile:5]: whats this for?

**Review by ice-ares — DISMISSED** [review]

**Review by ice-atlas — DISMISSED** [review]

**Review by ice-atlas — DISMISSED** [review]

File: fastlane/Matchfile (hunk 8 above)
Comment (ice-hector) [reply fastlane/Matchfile:5]: @ice-ares  When we make match:read (and other related fastlane script runs) locally it always asks us to fill manually the bundle ids. This setting prevents this, and passes this step automatically

**Review by ice-hector — APPROVED** [review]

**Review by ice-hector — DISMISSED** [review]

**Review by ice-atlas — DISMISSED** [review]

**Review by ice-hector — APPROVED** [review]

**Pull request merged**